safe-global/safe-wallet-web
# fix: [Counterfactual] Add event bus and pending state

## What it solves

Part of #3156

## How this PR fixes it

- Adds an Event Bus for `SafeCreationEvents`
- Dispatches `SafeCreationEvents` when deploying a counterfactual safe
- Persists pending state status in `undeployedSafesSlice`
- Shows notifications for `SafeCreationEvents`
- Monitors undeployedSafes if they are relaying or processing
- Checks for an undeployedSafe if its already deployed on-chain and cleans up local storage in case it is
- Supports cancellation and speed-up safe creations
- Disables the create button in the safe creation flow and adds a loading spinner while submitting

## How to test it

Happy path:

1. Create a counterfactual safe
2. Deploy safe via "Activate now" or with first tx
4. Observe Processing notification showing
5. Observe the "Activate now" button is disabled and has a loading state on it
8. Observe a Success notification
9. The undeployed state should be cleaned up

Pending state is persisted:

1. Create a counterfactual safe
2. Deploye safe via "Activate now" or with first tx
3. Observe Processing notification showing
4. Reload the app
5. Observe the "Activate now" button is disabled and has a loading state on it
6. Observe a Success notification

Sad path:

1. Create a counterfactual safe
2. Deploy safe via "Activate now" or with first tx with too little gas
3. Observe Processing notification
4. Observe Timeout error after a while
5. Cancel transaction in wallet and execute again
6. This time speed up the transaction
7. Observe a success notification

Cleaning state:

1. Create a counterfactual safe
2. Deploy the safe via "Activate now" and immediately close the app
3. Wait until the transaction is successfully executed via wallet
4. Open the app again
5. Observe the undeployed state is cleaned up

Creation Loading state:

1. Go through the safe creation flow
2. Press Create
3. Observe that the Create button is disabled and has a loading spinner

## Screenshots

## Checklist
